### PR TITLE
Clarify docs for some operators

### LINKED ISSUE #3922: The documentation for the `<=>` operator is missing
Except if I miss something, there is no documentation for the `<=>` operator even though it is mentioned in both Twig 2 and 3 documentation:

https://twig.symfony.com/doc/2.x/templates.html#expressions
https://twig.symfony.com/doc/3.x/templates.html#expressions

### PR DESCRIPTION
**Pull request by fabpot**
Closes #3922

### CODE REVIEW

File: doc/templates.rst [hunk 1]
```diff
@@ -727,31 +727,15 @@ You can combine multiple expressions with the following operators:
...
-    `regular expressions`_:
-
-    .. code-block:: twig
-
-        {% if phone matches '/^[\\d\\.]+$/' %}
-        {% endif %}
+Iterator Operators
+~~~~~~~~~~~~~~~~~~
 
 Check that a sequence or a mapping ``has every`` or ``has some`` of its
```
Comment (stof) [inline doc/templates.rst]: Maybe we should document that `has every` returns `true` for an empty iterable and `has some` returns `false`

Comment (stof) [inline doc/templates.rst]: should this talk about iterable instead of `sequence or mapping` ?

File: doc/templates.rst [hunk 2]
```diff
@@ -727,31 +727,15 @@ You can combine multiple expressions with the following operators:
 Comparisons
 ~~~~~~~~~~~
 
-The following comparison operators are supported in any expression: ``==``,
-``!=``, ``<``, ``>``, ``>=``, and ``<=``.
+The following mathematical comparison operators are supported in any
+expression: ``==``, ``!=``, ``<``, ``>``, ``>=``, ``<=``, and ``<=>``.
```
Comment (stof) [inline doc/templates.rst]: should the spaceship operator be considered a mathematical operator ?

File: doc/templates.rst (hunk 1 above)
Comment (fabpot) [reply doc/templates.rst]: Done

Comment (fabpot) [reply doc/templates.rst]: Done

File: doc/templates.rst (hunk 2 above)
Comment (fabpot) [reply doc/templates.rst]: I've created a specific section for the spaceship operator.

**Review by stof — APPROVED** [review]

**Pull request merged**